include/mqt-core/dd/Operations.hpp

```diff
@@ -34,83 +34,99 @@
 #include <vector>
 
 namespace dd {
-// single-target Operations
-template <class Config>
-qc::MatrixDD
-getStandardOperationDD(const qc::StandardOperation* op, Package<Config>& dd,
-                       const qc::Controls& controls, const qc::Qubit target,
-                       const bool inverse) {
-  GateMatrix gm;
 
-  const auto type = op->getType();
+inline std::pair<qc::OpType, std::vector<fp>>
+getOperationParameters(const qc::StandardOperation* op, qc::Qubit* target0,
+                       qc::Qubit* target1, bool inverse) {
+  auto type = op->getType();
   const auto& parameter = op->getParameter();
 
   switch (type) {
+  // self-inverse Operations
   case qc::I:
-    gm = I_MAT;
-    break;
   case qc::H:
-    gm = H_MAT;
-    break;
   case qc::X:
-    gm = X_MAT;
-    break;
   case qc::Y:
-    gm = Y_MAT;
-    break;
   case qc::Z:
-    gm = Z_MAT;
-    break;
+  case qc::SWAP:
+  case qc::ECR:
+    return {type, {}};
+  // inverse by using the inverse type
+  case qc::iSWAP:
+  case qc::iSWAPdg:
+  case qc::Peres:
+  case qc::Peresdg:
   case qc::S:
-    gm = inverse ? SDG_MAT : S_MAT;
-    break;
   case qc::Sdg:
-    gm = inverse ? S_MAT : SDG_MAT;
-    break;
   case qc::T:
-    gm = inverse ? TDG_MAT : T_MAT;
-    break;
   case qc::Tdg:
-    gm = inverse ? T_MAT : TDG_MAT;
-    break;
   case qc::V:
-    gm = inverse ? VDG_MAT : V_MAT;
-    break;
   case qc::Vdg:
-    gm = inverse ? V_MAT : VDG_MAT;
-    break;
-  case qc::U:
-    gm = inverse ? uMat(-parameter[1U], -parameter[2U], -parameter[0U])
-                 : uMat(parameter[2U], parameter[1U], parameter[0U]);
-    break;
-  case qc::U2:
-    gm = inverse ? u2Mat(-parameter[0U] + PI, -parameter[1U] - PI)
-                 : u2Mat(parameter[1U], parameter[0U]);
-    break;
-  case qc::P:
-    gm = inverse ? pMat(-parameter[0U]) : pMat(parameter[0U]);
-    break;
   case qc::SX:
-    gm = inverse ? SXDG_MAT : SX_MAT;
-    break;
   case qc::SXdg:
-    gm = inverse ? SX_MAT : SXDG_MAT;
-    break;
+    if (inverse) {
+      type = static_cast<qc::OpType>(+type ^ qc::OpTypeInv);
+    }
+    return {type, {}};
+  // inverse by negating the parameter
+  case qc::RXX:
+  case qc::RYY:
+  case qc::RZZ:
+  case qc::RZX:
   case qc::RX:
-    gm = inverse ? rxMat(-parameter[0U]) : rxMat(parameter[0U]);
-    break;
   case qc::RY:
-    gm = inverse ? ryMat(-parameter[0U]) : ryMat(parameter[0U]);
-    break;
   case qc::RZ:
-    gm = inverse ? rzMat(-parameter[0U]) : rzMat(parameter[0U]);
-    break;
+  case qc::P:
+    if (inverse) {
+      return {type, {-parameter[0U]}};
+    }
+    return {type, {parameter[0U]}};
+  // inverse by negating the first of two parameter
+  case qc::XXminusYY:
+  case qc::XXplusYY:
+    if (inverse) {
+      return {type, {-parameter[0U], parameter[1U]}};
+    }
+    return {type, {parameter[0U], parameter[1U]}};
```

Comment:
This comment isn't fully true.

Reply:
Why not? These are the gates with to parameters that can be inverted by inverting the first parameter? Am i missing something?

Reply:
That's my bad. I must have read this too quickly. The comments seems just fine 🤨